"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const vitest_1 = require("vitest");
const checkIsResponseError_1 = require("./checkIsResponseError");
(0, vitest_1.describe)('checkIsResponseError', () => {
    (0, vitest_1.it)('should return true for a valid IResponseError object', () => {
        const error = {
            error: 'Something went wrong',
            message: 'An error occurred',
            statusCode: 500,
        };
        const result = (0, checkIsResponseError_1.checkIsResponseError)(error);
        (0, vitest_1.expect)(result).toBe(true);
    });
    (0, vitest_1.it)('should return false for null', () => {
        const result = (0, checkIsResponseError_1.checkIsResponseError)(null);
        (0, vitest_1.expect)(result).toBe(false);
    });
    (0, vitest_1.it)('should return false for undefined', () => {
        const result = (0, checkIsResponseError_1.checkIsResponseError)(undefined);
        (0, vitest_1.expect)(result).toBe(false);
    });
    (0, vitest_1.it)('should return false for a non-object value', () => {
        const result = (0, checkIsResponseError_1.checkIsResponseError)('This is a string');
        (0, vitest_1.expect)(result).toBe(false);
    });
    (0, vitest_1.it)('should return false if the object is missing the "error" property', () => {
        const error = {
            message: 'An error occurred',
            statusCode: 500,
        };
        const result = (0, checkIsResponseError_1.checkIsResponseError)(error);
        (0, vitest_1.expect)(result).toBe(false);
    });
    (0, vitest_1.it)('should return false if the object is missing the "message" property', () => {
        const error = {
            error: 'Something went wrong',
            statusCode: 500,
        };
        const result = (0, checkIsResponseError_1.checkIsResponseError)(error);
        (0, vitest_1.expect)(result).toBe(false);
    });
    (0, vitest_1.it)('should return false if the object is missing the "statusCode" property', () => {
        const error = {
            error: 'Something went wrong',
            message: 'An error occurred',
        };
        const result = (0, checkIsResponseError_1.checkIsResponseError)(error);
        (0, vitest_1.expect)(result).toBe(false);
    });
});
